docker/mkimage

```diff
@@ -33,6 +33,11 @@ sudo_if_needed docker build \
   --file=${DOCKER_FILE} \
   --cache-from="${DOCKER_IMAGE_NAME}:${DOCKER_VERSION}" \
   --tag="${DOCKER_IMAGE_NAME}:${DOCKER_VERSION}" \
+  --build-arg=FROM_IMAGE_FOR_SOLVERS=rvt_base:latest \
+  --build-arg=FROM_IMAGE_FOR_KLEE=rvt_solvers:latest \
+  --build-arg=FROM_IMAGE_FOR_SEAHORN=rvt_klee:latest \
+  --build-arg=FROM_IMAGE_FOR_RVT=rvt_seahorn:latest \
+  --build-arg=FROM_IMAGE_FOR_RVT_R2CT=rvt:latest \
   --build-arg=USERNAME="$DOCKER_USER" \
   --build-arg=USER_UID="$DOCKER_UID" \
   --build-arg=USER_GID="$DOCKER_GID" \
```

Comment:
This is already quite a big PR, and I think that building images as part of CI is a more important first step than fiddling with permissions, so maybe this is probably something to put on the backlog rather than fixing right now.

I checked into how `cross` does things a while back, and wrote this:

https://github.com/project-oak/rust-verification-tools/issues/139#issuecomment-847872988

(I actually came to this repo looking for that comment for an unrelated project, and stumbled on this PR. Nice work)

Reply:
Indeed, I'm sorry about the verbosity. 

I can split this PR into smaller new ones easier to deal with if you'd like.

Reply:
> Indeed, I'm sorry about the verbosity.

This PR is about as big as I was expecting this pull request to be (which is part of the reason why I decided not to try writing it ;-D )

I think you've made the right choice about how much to include in this PR. Hopefully the maintainer agrees, and merges it next week.

Reply:
Regarding the permissions-related build arguments, this PR didn't introduce them though 🤷🏽